Data.Rmd

```diff
@@ -0,0 +1,329 @@
+    This technique is not as common or well-known as those above, but can be very useful in specific situations.
+
+-   If you want to store data persistently across R sessions, such as configuration or user-specific data, there are officially sanctioned locations for this.
+
+## Exported data {#sec-data-data}
+
+The most common location for package data is (surprise!) `data/`.
+We recommend that each file in this directory be an `.rda` file created by `save()` containing a single R object, with the same name as the file.
+The easiest way to achieve this is to use `usethis::use_data()`.
+
+```{r, eval = FALSE}
+my_pkg_data <- sample(1000)
+usethis::use_data(my_pkg_data)
+```
+
+Let's imagine we are working on a package named "pkg".
+The snippet above creates `data/my_pkg_data.rda` inside the source of the pkg package and adds `LazyData: true` in your `DESCRIPTION`.
+This makes the `my_pkg_data` R object available to users of pkg via `pkg::my_pkg_data` or, after attaching pkg with `library(pkg)`, as `my_pkg_data`.
+
+The snippet above is something the maintainer executes once (or every time they need to update `my_pkg_data`).
+This is workflow code and should **not** appear in the `R/` directory of the source package.
+(We'll talk about a suitable place to keep this code below.) For larger datasets, you may want to experiment with the compression setting, which is under the control of the `compress` argument.
+The default is "bzip2", but sometimes "gzip" or "xz" can create smaller files.
+
+It's possible to use other types of files below `data/`, but we don't recommend it because `.rda` files are already fast, small, and explicit.
+The other possibilities are described in the documentation for `utils::data()` and in the [Data in packages](https://rstudio.github.io/r-manuals/r-exts/Creating-R-packages.html#data-in-packages) section of Writing R Extensions.
+In terms of advice to package authors, the help topic for `data()` seems to implicitly make the same recommendations as we do above:
+
+-   Store one R object in each `data/*.rda` file
+-   Use the same name for that object and its `.rda` file
+-   Use lazy-loading, by default
+
+If the `DESCRIPTION` contains `LazyData: true`, then datasets will be lazily loaded.
+This means that they won't occupy any memory until you use them.
+The following example shows memory usage before and after loading the nycflights13 package.
+You can see that memory usage doesn't change until you inspect the `flights` dataset stored inside the package.
+
+```{r}
+pryr::mem_used()
+library(nycflights13)
+pryr::mem_used()
+
+invisible(flights)
+pryr::mem_used()
+```
+
+We recommend that you include `LazyData: true` in your `DESCRIPTION` if you are shipping `.rda` files below `data/`.
+If you use `use_data()` to create such datasets, it will automatically make this modification to `DESCRIPTION` for you.
+
+In terms of advice to package users, note that lazily-loaded datasets do **not** need to be pre-loaded.
```

Comment:
Somehow pull this out in a call out that acknowledges the historical important of `data()`, but that you should never use it now.

Reply:
done